pkgs/applications/window-managers/cagebreak/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub
+, meson, ninja, pkg-config, wayland, scdoc, makeWrapper
+, wlroots, wayland-protocols, pixman, libxkbcommon
+, cairo , pango, fontconfig, pandoc, udev
+, xwayland ? null
+, nixosTests
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cagebreak";
+  version = "1.4.2";
+
+  src = fetchFromGitHub {
+    owner = "project-repo";
+    repo = "cagebreak";
+    rev = version;
+    sha256 = "+Ww1rsUR7qe/BixLPR8GiRc3C6QmpLzWpT2wym8b4/M=";
+  };
+
+  nativeBuildInputs = [ meson ninja pkg-config wayland scdoc makeWrapper ];
+
+  buildInputs = [
+    wlroots wayland wayland-protocols pixman libxkbcommon cairo
+    pango.dev fontconfig pandoc udev
+  ];
+
+  outputs = [ "out" "contrib" ];
+
+  mesonFlags = [
+    "-Dxwayland=${stdenv.lib.boolToString (xwayland != null)}"
+    "-Dversion_override=${version}"
+  ];
+
+  mesonBuildType = "release";
+
+  postInstall = ''
+    mkdir -p $contrib/share/cagebreak
+    cp $src/examples/config $contrib/share/cagebreak/config
+  '';
+
+  postFixup = stdenv.lib.optionalString (xwayland != null) ''
```

Comment:
do you know where in the code it wants to call the binary? It would be best if they didn't try to look for it from `PATH` since it's done with conditional compilation anyways (and not just a runtime optional feature).
They could use `find_program` for it and then generate a header.

Reply:
The xwayland server is started by wlroots' libraries so no calls to the xserver binary take place in cagebreak.
The sway module requires xwayland in its `extraPackages` otherwise it fails to start the xwayland backend.